.github/actions/prepare-env/action.yml

```
@@ -0,0 +1,114 @@
+name: 'Prepare Environment'
+description: 'Sets up Docker, Configs, and Database for CI jobs'
+inputs:
+  install_host_python_deps:
+    description: 'Whether to install python dependencies on the host runner'
+    required: false
+    default: 'false'
+  extra_python_requirements:
+    description: 'Path to additional requirements file to install on host'
+    required: false
+    default: ''
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Install system tools
+      shell: bash
+      run: |
+        sudo apt-get update -y && sudo apt-get install -y wait-for-it retry
+
+    - name: Download the Docker containers
+      shell: bash
+      run: docker compose pull
+
+    - name: Clean up existing Docker volumes
+      shell: bash
+      run: |
+        docker compose down -v || true
+        docker volume rm -f omegaup_omegaupdata omegaup_dbdata omegaup_rabbitmq omegaup_redis || true
+
+    - name: Start the Docker containers
+      shell: bash
+      run: |
+        docker compose up --no-build --detach
+
+        # Add aliases
+        cat << EOF | sudo tee --append /etc/hosts
+        127.0.0.1 mysql grader runner gitserver broadcaster redis rabbitmq
+        EOF
+
+        # Wait for common services
+        wait-for-it -t 60 mysql:13306
+        wait-for-it -t 60 gitserver:33861
+        wait-for-it -t 60 redis:6379
+
+        # Wait for mysql to be fully ready
+        for i in $(seq 30); do
+          if [[ "$(mysql -uomegaup -pomegaup --host=mysql --port=13306 --skip-column-names --batch -e 'SELECT 1;' || echo 0)" == "1" ]]; then
```

Could you verify that we actually have access to the `omegaup` database instead of just running `SELECT 1`? 

```suggestion
          if [[ "$(mysql -uomegaup -pomegaup --host=mysql --port=13306 --skip-column-names --batch -e 'USE omegaup; SELECT 1;' || echo 0)" == "1" ]]; then
```